core/src/main/scala/org/http4s/parser/AdditionalRules.scala

```diff
@@ -213,4 +214,11 @@ private[http4s] object AdditionalRules {
   def EOI = P.char('\uFFFF')
 
   def EOL = optWs *> EOI.rep
+
+  val Digits: Parser1[Long] = P.rep1(Rfc3986.digit, 1).string.mapFilter { s =>
```

Comment:
There's a `digit` in `cats.parse.Rfc5234` that we can use.  I guess the value in having a separate rule is that yours does the work of parsing a run of them to a long, while the cats-parse one just gives you the Char.

I think I'd call this `Long`, to show that arbitrary lengths will fail. Or maybe even `NonNegativeLong`, because a leading minus won't work here.  I do think it's useful.